### PR TITLE
feat: move typst-preview to tinymist

### PR DESCRIPTION
**Pull request by Myriad-Dreamin**
This PR doesn't change any functional code but moves all of them to tinymist side.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -167,11 +173,22 @@ jobs:
         with:
           name: tinymist-${{ env.target }}
           path: tinymist-${{ env.target }}${{ fromJSON('["", ".exe"]')[matrix.platform == 'win32'] }}
-      - name: Package extension
+      - name: Package typst-preview extension
+        if: (startsWith(github.ref, 'refs/tags/') || matrix.regular_build == 'true')
```
Comment (Enter-tainer) [inline .github/workflows/release.yml:177]: Is it only build when tags are pushed? Is this expected?

File: contrib/typst-preview/editors/vscode/package.json [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+      {
+        "command": "typst-preview.preview",
+        "key": "ctrl+k v",
+        "mac": "cmd+k v",
+        "when": "editorLangId == typst"
+      }
+    ]
+  },
+  "scripts": {
+    "build:frontend": "cd ../../../../tools/typst-preview-frontend && yarn run build && cd ../../contrib/typst-preview/editors/vscode && rimraf ./out/frontend/ && cpr ../../../../tools/typst-preview-frontend/dist/ ./out/frontend/ && rimraf ../../../../crates/tinymist-assets/src/typst-preview.html && cpr ./out/frontend/index.html ../../../../crates/tinymist-assets/src/typst-preview.html",
```
Comment (Enter-tainer) [inline contrib/typst-preview/editors/vscode/package.json:254]: okay, magic happens here

**Review by Enter-tainer — COMMENTED** [review]
Overall LGTM. There is still something left:

1. How to release typst-preview after the merge? I guess I need add token to somewhere to make it work?
2. Please add typst-preview's contributors to the [co-author](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors) of certain commit. Namely, these 16 people: https://github.com/Enter-tainer/typst-preview/graphs/contributors

File: .github/workflows/release.yml (hunk 1 above)
Comment (Myriad-Dreamin) [reply .github/workflows/release.yml:177]: Either it is tagged or marked as `regular_build` in the matrix (Linux x86, Windows x86, mac ARM)

File: contrib/typst-preview/editors/vscode/package.json (hunk 2 above)
Comment (Myriad-Dreamin) [reply contrib/typst-preview/editors/vscode/package.json:254]: Looks not great, but I just changed the relative paths and doesn't change any other thing.

**Comment by Myriad-Dreamin** [comment]
> 1. How to release typst-preview after the merge? I guess I need add token to somewhere to make it work?
> 2. Please add typst-preview's contributors to the [co-author](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors) of certain commit. Namely, these 16 people: https://github.com/Enter-tainer/typst-preview/graphs/contributors. You may want to let GPT write some script to do that for you.

1. I'm not sure, but it is possible to let typst-preview repository repository continue releasing and documenting it, and the token will be still only visible and mantained by you. This is up to you.
2. will do before merging.

**Comment by Enter-tainer** [comment]
> I'm not sure, but it is possible to let typst-preview repository repository continue releasing and documenting it

I guess it will download artifact from tinymist repo then?

> the token will be still only visible and mantained by you. This is up to you.

I think adding the token to tinymist is not a big deal as long as i can do it myself. Currently I cannot see the "setting" panel maybe because of insufficient permission.

I personally feels that if we can fully integrate typst-preview into tinymist within a short period(like, before typst 0.12.0), we can still release typst-preview in original repo at this moment. And after it is fully integrated, user will not need typst-preview extension anymore.

If that's not possible, I prefer adding my tokens here and we can also port release scripts.

File: .github/workflows/release.yml (hunk 1 above)
Comment (Enter-tainer) [reply .github/workflows/release.yml:177]: make sense

**Comment by Myriad-Dreamin** [comment]
> 2. Please add typst-preview's contributors to the [co-author](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors) of certain commit. Namely, these 16 people: https://github.com/Enter-tainer/typst-preview/graphs/contributors. You may want to let GPT write some script to do that for you.

Done.

**Review by Enter-tainer — APPROVED** [review]
LGTM

**Pull request merged**